sawtooth/client.py

```diff
@@ -779,24 +779,34 @@ def get_transaction_status(self, txnid, timeout=30):
             request.get_method = lambda: 'HEAD'
             opener = urllib2.build_opener(self.proxy_handler)
             response = opener.open(request, timeout=timeout)
-            code = response.getcode()
-            response.close()
-            if code == 200:
-                return transaction.Status.committed
-            elif code == 302:
-                return transaction.Status.pending
-            else:
-                return transaction.Status.unknown
 
         except urllib2.HTTPError as err:
             LOGGER.error('peer operation on url %s failed with response: %d',
                          url, err.code)
+            if err.code == 302:
+                return transaction.Status.pending
+
+            raise MessageException('operation failed '
+                                   'with response: {0}'.format(err.code))
 
         except urllib2.URLError as err:
             LOGGER.error('peer operation on url %s failed: %s', url,
                          err.reason)
+            raise MessageException('operation failed: {0}'.format(err.reason))
+
+        except:
```

Comment:
In general, we should avoid bare excepts.  This bare except would catch all exceptions, including coding errors (typos and such), and return "no response from peer" which would be incorrect in those cases.